Sources/OAuth2Swift.swift

```diff
@@ -118,32 +124,24 @@ open class OAuth2Swift: OAuthSwift {
             }
         }
 
-        var queryString = "client_id=\(self.consumerKey)"
-        if encodeCallbackURL {
-            queryString += "&redirect_uri=\(callbackURL.absoluteString.urlEncodedString)"
-        } else {
-            queryString += "&redirect_uri=\(callbackURL.absoluteString)"
-        }
-        queryString += "&response_type=\(self.responseType)"
-        if !scope.isEmpty {
-            queryString += "&scope=\(scope)"
-        }
-        if !state.isEmpty {
-            queryString += "&state=\(state)"
-        }
-        for param in parameters {
-            queryString += "&\(param.0)=\(param.1)"
-        }
+        var queryString: String? = ""
```

Comment:
To make this work, parameters are encoded as they are added, which may produce `nil` value, just like when you [encode the whole string](https://github.com/OAuthSwift/OAuthSwift/blob/b50109fd752f93c93151de4b7695bd710aa26bed/Sources/OAuth2Swift.swift#L141). Here we add them and if everything is ok, we end up with the same encoded query string, if not – it's `nil`, which we [check in the same fashion](https://github.com/ianbytchek/OAuthSwift/blob/3915fcace18ede9bf21d7767a16ac80df967db96/Sources/OAuth2Swift.swift#L139) and [fail with the same error](https://github.com/ianbytchek/OAuthSwift/blob/3915fcace18ede9bf21d7767a16ac80df967db96/Sources/OAuth2Swift.swift#L144).